### PR TITLE
statistics: migrate test-infra to testify for statistics/handle/dump_test.go

### LINKED ISSUE #28130: migrate test-infra to testify for `statistics/handle/dump_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28130

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
Nice work! LGTM.

File: statistics/handle/dump_test.go [hunk 1]
```diff
@@ -89,49 +138,51 @@ func (s *testStatsSuite) TestDumpGlobalStats(c *C) {
...
+	clearedStats := getStatsJSON(t, dom, "test", "t")
+	require.Equal(t, 0, len(clearedStats.Partitions))
 
 	// load global-stats back
-	c.Assert(s.do.StatsHandle().LoadStatsFromJSON(s.do.InfoSchema(), globalStats), IsNil)
-	loadedStats := s.getStatsJSON(c, "test", "t")
-	c.Assert(len(loadedStats.Partitions), Equals, 3) // p0, p1, global
+	require.Nil(t, dom.StatsHandle().LoadStatsFromJSON(dom.InfoSchema(), globalStats))
+	loadedStats := getStatsJSON(t, dom, "test", "t")
+	require.Equal(t, 3, len(loadedStats.Partitions))
```
Comment (xuyifangreeneyes) [inline statistics/handle/dump_test.go:179]: Add the comment `// p0, p1, global` so it's easier for others to understand the test

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

File: statistics/handle/dump_test.go (hunk 1 above)
Comment (feitian124) [reply statistics/handle/dump_test.go]: ok, updated

**Comment by feitian124** [comment]
could  you help merge? @mmyj

**Review by disksing — APPROVED** [review]

**Comment by feitian124** [comment]
> @feitian124: Your PR was out of date, I have automatically updated it for you.
> 
> At the same time I will also trigger all tests for you:
> 
> /run-all-tests
> 
> If the CI test fails, you just re-trigger the test that failed and the bot will merge the PR for you after the CI passes.

@zhouqiang-cl the bot behaves different from what it says, it requires another review instead of merge directly. 
this is different as before. Is this changed as expect or a bug?

**Pull request merged**